### PR TITLE
Json exporter fix

### PR DESCRIPTION
**Pull request by marcnet80**

### CODE REVIEW

File: src/BenchmarkDotNet/Exporters/Json/SimpleJson.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Exporters/Json/SimpleJson.cs:1575]: I wonder why we do get `NaN`s here. Most probably by doing division by `0` using doubles?

@marcnet80  In such a case, would it be better (and possible) to return an empty string instead of `0.0`? Just to say "there is no value" instead "the value is 0.0" (it would be hard to tell if there was actual 0 or NaN serialized as 0).

```suggestion
                    return string.Empty;
```

File: tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs
Comment (adamsitnik) [inline tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs:24]: we already have tests that check the Json serializer output: https://github.com/dotnet/BenchmarkDotNet/blob/9744089383f881506c8069f9802260e63c448ea1/tests/BenchmarkDotNet.Tests/Exporters/CommonExporterApprovalTests.cs#L85-L88

Could you please remove this test, make `ReplaceUnsupportedNumericValues` internal and just test what it returns for valid and invalid inputs?

**Review by adamsitnik — COMMENTED** [review]
Hi @marcnet80 

Big thanks for your PR!

Overall it looks good, but I have left some comments. Could you please address them before I merge the PR?

Thanks,
Adam

File: src/BenchmarkDotNet/Exporters/Json/SimpleJson.cs
Comment (marcnet80) [reply src/BenchmarkDotNet/Exporters/Json/SimpleJson.cs]: Hello, @adamsitnik. Yes,`NaN` occurs in result of doubles zero division here: https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Mathematics/Statistics.cs#L61, it calls Perfolizer method: https://github.com/AndreyAkinshin/perfolizer/blob/master/src/Perfolizer/Perfolizer/Mathematics/Common/Moments.cs#L31

Propose to return escaped empty string like `"\"\""`, because we have `StringBuilder`, in resulted Json will be: `""`.

File: tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs
Comment (marcnet80) [reply tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs]: Some tests will be failed due hardcoded `NaN` values in text files from ApprovedFiles folder.
Please, review new changes.

Comment (adamsitnik) [reply tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs]: @marcnet80 the new changes look good to me! but we need to get the CI green.

The easiest way of updating the approved files after on purpose exporter changes is:

* go to `tests\BenchmarkDotNet.Tests`
* run the tests `dotnet test -c Release -f netcoreapp2.1`
* go to `Exporters\ApprovedFiles` and check which `.approved` file is different than `.received`. Ensure that the changes are what we want and remove the old `.approved` file and rename the `.received` file `.approved` 
* re-run the tests
* if everything is green, push the changes

obraz

Comment (marcnet80) [reply tests/BenchmarkDotNet.Tests/Exporters/JsonExporterTests.cs]: fixed.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @marcnet80 !

**Pull request merged**
